nucypher/cli/deploy.py

```diff
@@ -171,42 +170,42 @@ def deploy(action,
 
     elif action == "contracts":
 
-        registry_filepath = deployer.blockchain.registry.filepath
-        if os.path.isfile(registry_filepath):
-            click.secho(f"\nThere is an existing contract registry at {registry_filepath}.\n"
-                        f"Did you mean 'nucypher-deploy upgrade'?\n", fg='yellow')
-            click.confirm("Optionally, destroy existing local registry and continue?", abort=True)
-            click.confirm(f"Confirm deletion of contract registry '{registry_filepath}'?", abort=True)
-            os.remove(registry_filepath)
-
         #
         # Deploy Single Contract
         #
 
         if contract_name:
-            deployment_secret = click.prompt(f"Enter deployment secret for {contract_name}", confirmation_prompt=True)
-
             try:
-                deployer.deployers[contract_name]
+                contract_deployer = deployer.deployers[contract_name]
             except KeyError:
                 message = f"No such contract {contract_name}. Available contracts are {deployer.deployers.keys()}"
                 click.secho(message, fg='red', bold=True)
                 raise click.Abort()
             else:
-                # Deploy single contract
-                receipts, agent = deployer.deploy_contract(contract_name=contract_name,
-                                                           plaintext_secret=deployment_secret)
+                click.secho(f"Deploying {contract_name}")
+                if contract_deployer._upgradeable:
+                    secret = deployer.collect_deployment_secret(deployer=contract_deployer)
+                    receipts, agent = deployer.deploy_contract(contract_name=contract_name, plaintext_secret=secret)
```

Comment:
```suggestion
                    receipts, agent = deployer.deploy_contract(contract_name=contract_name, plaintext_secret=secret, gas_limit=gas)
```
It also needs to be passed to `deployer.deploy_network_contracts(secrets=secrets)` in line 243:
https://github.com/nucypher/nucypher/pull/1131/files#diff-4809c009687ec0119c014880f31f9111R243